From 2b340fc84c292db1272ae4b3a7eb85a3de223ddd Mon Sep 17 00:00:00 2001
From: Dominique Martinet <dominique.martinet@atmark-techno.com>
Date: Tue, 2 Jul 2024 10:00:38 +0900
Subject: [PATCH] disable -fzero-call-used-regs=used on ppc64le

This fails as follow for some files:
packet.c: In function 'ssh_packet_log_type':
packet.c:1158:1: sorry, unimplemented: argument 'used' is not supported for '-fzero-call-used-regs' on this target
 1158 | }
      | ^
make: *** [Makefile:203: packet.o] Error 1

This had previously been an issue on an older version as well and the
"fix" at the time was to make the detection function more likely to
trigger that behaviour, but that was apparently not enough so just
disable at configure level.

Link: https://issues.guix.gnu.org/68212
---
 configure.ac | 12 ++++++++----
 1 file changed, 8 insertions(+), 4 deletions(-)

diff --git a/configure.ac b/configure.ac
index 5a865f8e1b07..4f99ad35a3ce 100644
--- a/configure.ac
+++ b/configure.ac
@@ -237,10 +237,14 @@ if test "$GCC" = "yes" || test "$GCC" = "egcs"; then
 	# https://github.com/llvm/llvm-project/issues/59242
 	# clang 17 has a different bug that causes an ICE when using this
 	# flag at all (https://bugzilla.mindrot.org/show_bug.cgi?id=3629)
-	case "$CLANG_VER" in
-	apple-15*) OSSH_CHECK_CFLAG_LINK([-fzero-call-used-regs=used]) ;;
-	17*)	;;
-	*)	OSSH_CHECK_CFLAG_LINK([-fzero-call-used-regs=used]) ;;
+	case "$host" in
+	"powerpc64le"*) ;; # skip on ppc64le
+	*)
+		case "$CLANG_VER" in
+		apple-15*) OSSH_CHECK_CFLAG_LINK([-fzero-call-used-regs=used]) ;;
+		17*)	;;
+		*)	OSSH_CHECK_CFLAG_LINK([-fzero-call-used-regs=used]) ;;
+		esac
 	esac
 	OSSH_CHECK_CFLAG_COMPILE([-ftrivial-auto-var-init=zero])
     fi
-- 
2.39.2

